searchcore/src/vespa/searchcore/proton/server/documentretriever.cpp

```diff
@@ -109,12 +109,13 @@ ::DocumentRetriever(const DocTypeName &docTypeName,
                 && ((*attr)->getBasicType() != BasicType::PREDICATE)
                 && ((*attr)->getBasicType() != BasicType::REFERENCE))
             {
-                _attributeFields.insert(field);
+                _attributeFields.emplace_back(field);
             } else {
                 _areAllFieldsAttributes = false;
             }
         }
     }
+    std::sort(_attributeFields.begin(), _attributeFields.end(), document::Field::FieldPtrComparator());
```

Comment:
consider using a builder to help you make legal field sets. Also consider having a small field set wrapper around the vector to signal to the world that this vector is a set.